x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1190,6 +1345,26 @@ private static ElasticsearchSecurityException invalidGrantException(String detai
         return exception;
     }
 
+    /**
+     * Logs an exception at TRACE level (if enabled)
+     */
+    private <E extends Throwable> E traceLog(String action, E exception) {
+        if (logger.isTraceEnabled()) {
+            if (exception instanceof ElasticsearchException) {
+                final ElasticsearchException esEx = (ElasticsearchException) exception;
+                final Object detail = esEx.getHeader("error_description");
+                if (detail != null) {
+                    logger.trace("Failure in [{}] - [{}] [{}]", action, detail, esEx.getDetailedMessage());
```

Comment:
Lets log the exception itself instead of the detailed message so we get the stacktrace